### PR TITLE
Use rustup rust-analyzer component when there is a toolchain file override for the opened workspace

### PR DESCRIPTION
**Pull request by Veykril**
Fixes https://github.com/rust-lang/rust-analyzer/issues/17663

### CODE REVIEW

File: editors/code/src/bootstrap.ts [hunk 1]
```diff
@@ -51,12 +52,29 @@ async function getServer(
     }
     if (packageJson.releaseTag === null) return "rust-analyzer";
 
+    if (vscode.workspace.workspaceFolders?.length === 1) {
+        // otherwise check if there is a toolchain override for the current vscode workspace
+        // and if the toolchain of this override has a rust-analyzer component
+        // if so, use the rust-analyzer component
+        const toolchainTomlExists = await fileExists(
+            vscode.Uri.joinPath(vscode.workspace.workspaceFolders[0]!.uri, "rust-toolchain.toml"),
```
Comment (lnicola) [inline editors/code/src/bootstrap.ts:60]: Nb. this doesn't handle `rust-toolchain`, but I think it's fine.

**Comment by davidbarsky** [comment]
Wait, why does this one prefer a rustup-provided rust-analyzer over a bundled rust-analyzer? I'd expect that the bundled one would take priority.

**Comment by Veykril** [comment]
See the discussion in https://github.com/rust-lang/rust-analyzer/issues/17663, if the user explicitkly installed the rust-analyzer component in a toolchain override it is a big indicator they meant to use that instead. And a toolchain r-a is usually compatible with the toolchain where as the bundled one only supports the last couple stable releases

**Comment by davidbarsky** [comment]
Ah, I assumed that the presence of any `rust-toolchain.toml` file would pull the rust-analyzer component from there. My bad!

**Pull request merged**